gui/pregnancy.lua

```diff
@@ -0,0 +1,328 @@
+local gui = require('gui')
+local widgets = require('gui.widgets')
+
+PregnancyGui = defclass(PregnancyGui, widgets.Window)
+PregnancyGui.ATTRS {
+    frame_title='My Window',
+    frame={w=50, h=45},
+    resizable=true, -- if resizing makes sense for your dialog
+    resize_min={w=50, h=20}, -- try to allow users to shrink your windows
+}
+
+function PregnancyGui:init()
+    self.mother = false
+    self.father = false
+    self.father_historical = false
+    self.msg = {}
+    -- self.success = false
+    self:addviews{
+        widgets.ResizingPanel{
+            frame={t=0},
+            frame_style=gui.FRAME_INTERIOR,
+            autoarrange_subviews=true,
+            subviews={
+                widgets.WrappedLabel{
+                    text_to_wrap=self:callback('getMotherLabel')
+                },
+                widgets.HotkeyLabel{
+                    frame={l=0},
+                    label="Select Mother",
+                    key='CUSTOM_SHIFT_M',
+                    on_activate=self:callback('selectmother'),
+                },
+            },
+        },
+        widgets.ResizingPanel{
+            frame={t=6},
+            frame_style=gui.FRAME_INTERIOR,
+            autoarrange_subviews=true,
+            subviews={
+                widgets.WrappedLabel{
+                    text_to_wrap=self:callback('getFatherLabel')
+                },
+                widgets.HotkeyLabel{
+                    frame={l=0},
+                    label="Select Father",
+                    key='CUSTOM_SHIFT_F',
+                    on_activate=self:callback('selectfather'),
+                },
+                widgets.HotkeyLabel{
+                    frame={l=5},
+                    label="Set Mother's spouse as the Father",
+                    key='CUSTOM_F',
+                    on_activate=self:callback('spouseFather'),
+                    disabled=function() return not self.mother or self.mother.relationship_ids.Spouse == -1 end
+                },
+            },
+        },
+        widgets.ResizingPanel{
+            frame={t=12},
+            frame_style=gui.FRAME_INTERIOR,
+            autoarrange_subviews=1,
+            subviews={
+                widgets.HotkeyLabel{
+                    frame={l=0},
+                    key='CUSTOM_SHIFT_P',
+                    label="Create pregnancy",
+                    on_activate=self:callback('CreatePregnancy'),
+                    enabled=function() return self.mother or self.father and self.father_historical end
```

Comment:
this condition doesn't make sense to me. it's enabled if the mother is set OR if the (father and father_historical) is set? How can you create a pregnancy without a mother being set?